### PR TITLE
Reset diagnostics

### PR DESCRIPTION
**Pull request by Razzeee**
Fixes https://github.com/elm-tooling/elm-language-server/issues/357

### CODE REVIEW

File: src/providers/diagnostics/diagnosticsProvider.ts [hunk 1]
```diff
@@ -200,11 +200,13 @@ export class DiagnosticsProvider {
     const text = document.getText();
 
     if (isSaveOrOpen) {
-      const elmMakeDiagnostics = await this.elmMakeDiagnostics.createDiagnostics(
+      const diagnosticList = await this.elmMakeDiagnostics.createDiagnostics(
```
Comment (Razzeee) [inline src/providers/diagnostics/diagnosticsProvider.ts:203]: Didn't mean to change this

File: src/providers/diagnostics/diagnosticsProvider.ts [hunk 2]
```diff
@@ -200,11 +200,13 @@ export class DiagnosticsProvider {
...
         uri,
       );
 
-      elmMakeDiagnostics.forEach((diagnostics, diagnosticsUri) => {
+      this.resetDiagnosticsForType(diagnosticList, DiagnosticKind.ElmMake);
```
Comment (Razzeee) [inline src/providers/diagnostics/diagnosticsProvider.ts:207]: Might be nicer to just push `[]` to elmMakeDiagnostics everything else should work fine then.

**Comment by jmbockhorst** [comment]
Yeah this seems probably fine for now. Would have some problems with multiple entry points but that's pretty complicated.

**Comment by Razzeee** [comment]
Added this for elm-analyse and elm-make now. I think I'm happy with the code, not sure how to add it for type inference diagnostics/if it needs it at all.

We need to test this a bit.

**Comment by Razzeee** [comment]
My tests seemed successful. I only tested elm-make and elm-analyse.

@bengolds can you run this branch and test it for a while too?

**Comment by bengolds** [comment]
So far, this is MILES better. No more stale errors yet!

**Pull request merged**